code/game/objects/items/stacks/tiles/tile_types.dm

```diff
@@ -299,6 +299,623 @@
 /obj/item/stack/tile/carpet/donk/thirty
 	amount = 30
 
+/obj/item/stack/tile/carpet/neon
+	name = "neon carpet"
+	singular_name = "neon carpet tile"
+	desc = "A piece of rubbery mat inset with a phosphorescent pattern."
+	inhand_icon_state = "tile-neon"
+	turf_type = /turf/open/floor/carpet/neon
+	merge_type = /obj/item/stack/tile/carpet/neon
+
+	// Neon overlay
+	/// The icon used for the neon overlay and emissive overlay.
+	var/neon_icon
+	/// The icon state used for the neon overlay and emissive overlay.
+	var/neon_icon_state
+	/// The icon state used for the neon overlay inhands.
+	var/neon_inhand_icon_state
+	/// The color used for the neon overlay.
+	var/neon_color
+	/// The alpha used for the emissive overlay.
+	var/emissive_alpha = 255
+
+/obj/item/stack/tile/carpet/neon/update_overlays()
+	. = ..()
+	var/mutable_appearance/neon_overlay = mutable_appearance(neon_icon || icon, neon_icon_state || icon_state, )
+	neon_overlay.color = neon_color
+	. += neon_overlay
+	. += emissive_appearance(neon_icon || icon, neon_icon_state || icon_state, alpha=emissive_alpha)
+
+/obj/item/stack/tile/carpet/neon/worn_overlays(isinhands, icon_file)
+	. = ..()
+	if(!isinhands || !neon_inhand_icon_state)
+		return
+	var/mutable_appearance/neon_overlay = mutable_appearance(icon_file, neon_inhand_icon_state, alpha=src.alpha)
+	neon_overlay.color = neon_color
+	. += neon_overlay
+	// TODO: Figure out how to add an emissive overlay to the inhand.
```

Comment:
is this TODO now

Reply:
I'm still trying to figure that one out. Adding the emissive overlay normally just results in it being squished down to the same plane as the rest of the in hand overlays.

Reply:
lmk when you figure it out or give up

Reply:
It's done.